packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts

```diff
@@ -0,0 +1,253 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {APIGatewayProxyEvent, APIGatewayProxyResult} from "aws-lambda"
+import {Logger} from "@aws-lambda-powertools/logger"
+import {injectLambdaContext} from "@aws-lambda-powertools/logger/middleware"
+import {DynamoDBClient, PutItemCommand} from "@aws-sdk/client-dynamodb"
+import {marshall} from "@aws-sdk/util-dynamodb"
+import middy from "@middy/core"
+import inputOutputLogger from "@middy/input-output-logger"
+import errorHandler from "@nhs/fhir-middy-error-handler"
+import {v4 as uuidv4} from "uuid"
+
+const logger = new Logger({serviceName: "updatePrescriptionStatus"})
+const client = new DynamoDBClient({region: "eu-west-2"})
+const tableName = process.env.TABLE_NAME
+
+interface DynamoDBItem {
+  RequestID: string | undefined;
+  PrescriptionID: string;
+  PatientNHSNumber: string;
+  PharmacyODSCode: string;
+  TaskID: string;
+  LineItemID: string;
+  TerminalStatusIndicator: string;
+  LastModified: string;
+  Timestamp: string;
+  RequestMessage: any;
+}
+
+const lambdaHandler = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const xRequestId = event.headers["x-request-id"]
+    let requestBody
+    try {
+      requestBody = JSON.parse(event.body || "")
+    } catch (jsonParseError) {
+      logger.error("Error parsing JSON", {jsonParseError: jsonParseError})
+      const errorResponseBody = {
+        resourceType: "OperationOutcome",
+        issue: [
+          {
+            code: "value",
+            severity: "error",
+            details: {
+              coding: [
+                {
+                  system: "https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode",
+                  code: "INVALID_JSON",
+                  display: "Invalid JSON format in request body"
+                }
+              ]
+            }
+          }
+        ]
+      }
+      return {
+        statusCode: 400,
+        body: JSON.stringify(errorResponseBody),
+        headers: {
+          "Content-Type": "application/fhir+json",
+          "Cache-Control": "no-cache"
+        }
+      }
+    }
+
+    const entries = requestBody.entry
+
+    if (!entries || entries.length === 0) {
+      logger.error("Missing required fields")
```

Comment:
This log line and the following response are still a bit misleading, rather than missing required fields the problem is a bit more fundamental that no entries have been provided.

This may be something for discussion, but this scenario where the rest of the request is valid but it simply contains no entries may warrant a 200 - OK - nothing to process type response instead.

Reply:
We have 200 status code if there are no entries to process.